### PR TITLE
Cert issues + default changes

### PR DESCRIPTION
**Pull request by rushtehrani**
This PR is WIP but it currently addresses these issues:
- Wildcard certificate is generated now.
- The gateway patch adds HTTPs and HTTP redirect support.
- Fixes incorrect labels for Istio
- Update default values with more specific information.
- Fix certificate flattened keys

### CODE REVIEW

**Comment by rushtehrani** [comment]
This PR is WIP but it currently addresses these issues:
- Wildcard certificate is generated now.
- The gateway patch adds HTTPs and HTTP redirect support.
- Fixes incorrect labels for Istio
- Update default values with more specific information.

File: cert-manager/base/certificate.yaml [hunk 1]
```diff
@@ -8,6 +8,6 @@ spec:
   issuerRef:
     name: cert-manager-acme-dns-issuer
     kind: ClusterIssuer
-  commonName: $(certManagerCommonName)
+  commonName: '$(certManagerCommonName)'
```
Comment (Vafilor) [inline cert-manager/base/certificate.yaml]: @rushtehrani Why do we need the single quotes here? Doesn't the result generate double quotes?

Comment (aleksandrmelnikov) [reply cert-manager/base/certificate.yaml]: I can answer that. If he doesn't add quotes, wildcard doesn't work.

Comment (aleksandrmelnikov) [reply cert-manager/base/certificate.yaml]: See: https://onepanelio.slack.com/archives/G9HV7FU9G/p1583545162266700

**Comment by aleksandrmelnikov** [comment]
👍

**Pull request merged**